app/models/api/v1/report.rb

```diff
@@ -70,18 +71,31 @@ def student_json(student)
     {
       id: student.id,
       name: student.name,
+      first_name: student.first_name,
+      last_name: student.last_name,
       started_offering:  started_offering
     }
   end
 
-  # Helpers that provide student answers grouped by embeddable key:
 
+  def self.update_report_learners(report_learners)
+    report_learners.each do |learner|
+      if learner.last_report.nil? or learner.last_report < MAX_REPORT_LEARNER_AGE.ago
+        learner.last_report = Time.now
+        learner.update_fields()
+      end
+    end
+  end
+
+  # Helpers that provide student answers grouped by embeddable key:
   def get_student_answers
     # Collect all the student answers for given offering.
     student_ids = @students.map { |s| s.id }
-    answers = Report::Learner.where(offering_id: @offering.id, student_id: student_ids).map do |report_learner|
+    report_learners = Report::Learner.where(offering_id: @offering.id, student_id: student_ids)
+    API::V1::Report.update_report_learners(report_learners)
```

Comment:
Yeah, this is still a bit hacky, and can possibly be removed. 

The rationale for this: Answers are serialized in the report learners. Because answers now have a feedback property, report-learners that haven't been created or updated since feedbacks were included have to be updated.

It would be better to run a migration I guess.

The client code also has a way of tolerating old answers without feedback.  But in this case, the client isn't able to add feedback to old serialized report learner answers. (Because it is missing the saveable id...)

The best thing to do would be to create and run a migration or script to accompany this format change.